src/esphome/cover/cover.h

```diff
@@ -22,12 +22,21 @@ enum CoverCommand {
   COVER_COMMAND_OPEN = 0,
   COVER_COMMAND_CLOSE,
   COVER_COMMAND_STOP,
+  COVER_COMMAND_POSITION,
+  COVER_COMMAND_TILT,
 };
 
+using position_value_t = float;
+using tilt_value_t = float;
```

Comment:
I think we should drop these typedefs. They're nice, but in C++ there's no way to enforce the use of them and you end up with something where everything is fragmented in parts where the typedef is used and parts where the base type is used directly. For example, with `position_closed_` etc below you use floats even though these typedefs exist, same goes for write_command.